src/lib/operators/join_index.cpp

```diff
@@ -49,15 +49,31 @@ bool JoinIndex::supports(const JoinConfiguration config) {
 JoinIndex::JoinIndex(const std::shared_ptr<const AbstractOperator>& left,
                      const std::shared_ptr<const AbstractOperator>& right, const JoinMode mode,
                      const OperatorJoinPredicate& primary_predicate,
-                     const std::vector<OperatorJoinPredicate>& secondary_predicates, const IndexSide index_side,
-                     const std::vector<ColumnID> pruned_column_ids)
+                     const std::vector<OperatorJoinPredicate>& secondary_predicates, const IndexSide index_side, const std::vector<ColumnID> pruned_column_ids)
     : AbstractJoinOperator(OperatorType::JoinIndex, left, right, mode, primary_predicate, secondary_predicates,
                            std::make_unique<JoinIndex::PerformanceData>()),
       _index_side(index_side),
-      _adjusted_primary_predicate(primary_predicate), _pruned_column_ids{pruned_column_ids} {
+      _adjusted_primary_predicate(primary_predicate) {
   if (_index_side == IndexSide::Left) {
     _adjusted_primary_predicate.flip();
   }
+
+  _index_column_id_before_pruning = _adjusted_primary_predicate.column_ids.second;
+  for (const auto pruned_column_id: pruned_column_ids) {
+    if (pruned_column_id <= _index_column_id_before_pruning) {
+      ++_index_column_id_before_pruning;
+    }
+    else {
+      break;
+    }
```

Comment:
You could increase the readability here by using a guard clause:
```
if (pruned_column_id > _index_column_id_before_pruning) {
  break;
}
++_index_column_id_before_pruning;

Reply:
Good idea, thanks!